Data/test.lua

```diff
@@ -0,0 +1,9646 @@
+      Abilities={
+        {
+          Category="Class",
+          Nature="AUTOMATIC",
+          Name="Bard",
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="1",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Bardic Knowledge",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_BardicKnowledge"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="1",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Bardic Countersong",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_CounterSong"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="1",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Bardic Performance",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_BardicPerformance"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="1",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Armored Casting",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_ArmoredCasting"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="1",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Cantrips",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_Cantrips"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="2",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Versatile Performance",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_VersatilePerformance"] == 0)
+            end,
+          },
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+    {
+      Level="2",
+      Abilities={
+        {
+          Category="Bard Class Feature",
+          Nature="AUTOMATIC",
+          Name="Bard ~ Well-Versed",
+          Conditions={
+            function (character)
+              return (character.Variables["Bard_CF_WellVersed"] == 0)
```

Comment:
Hrm.  I wonder if we can generalize this a bit.  All these "I have these ability unless I traded it away" seems like a lot of duplication, causing a _toooon_ of duplicated code just all over the something.  Maybe make "I replaced this feature" be built in rather than encoded in _literally_ every class ability, and having to know what magic name of the variable they picked is. :-)

Reply:
Actually, yeah. Lines 613 - 624 are just "at level 2, you get an ability called "Bard ~ Well-Versed"....  Hrm... I wonder what a good balance is between naively translating the LST and making these files useful themselves is.  If the files aren't useful themselves, why do they exist (if no one is ever meant to edit them, we could just load the LST files directly, for example).

I feel like it would be nicer to edit/validate the LUA, which means I think it _should_ be a goal that they be editable, which means we should try to remove redundancies where we can, so that I can fit more than one ability per screen.